.github/workflows/build.yml

```diff
@@ -201,7 +202,7 @@ jobs:
 
       - # === Deploy for Integration Tests # NEVER run this against production branches. This is meant for PR deployments. ===
         name: Deploy for Integration Tests # NEVER run this against production branches. This is meant for PR deployments.
-        if: github.ref != 'refs/heads/release'
+        if: github.ref != 'refs/heads/release' && github.ref != 'refs/heads/master'
```

Comment:
This feels wrong, now this job has implicit logic around nightly integration tests that might end up biting us later. You can use the conditional you used for the job on line 229 inside the run logic instead.

Reply:
Okay, I'll merge the two separate integration test steps into one.